### ISSUE TITLE
rror updating Grohe Blue data: 'dict' object has no attribute 'timestamp'

### ISSUE THREAD

**Issue by weaseel** [Flo-Schilli/homeassistant-grohe_sense#43]
Hello,

I have a grohe blue home, installed the component and set it up. All values are "unknown" and I get the following error:

ERROR (MainThread) [custom_components.grohe_sense.entities.grohe_blue_update_coordinator] Error updating Grohe Blue data: 'dict' object has no attribute 'timestamp'

This is the result of Grohe Sense: Get Dashboard Data

`locations:
  - id: 123456789
    name: ABCDEFG
    type: 2
    role: owner
    timezone: Europe/Berlin
    water_cost: -1
    energy_cost: -1
    heating_type: -1
    currency: EUR
    default_water_cost: 0.004256
    default_energy_cost: 0.003977
    default_heating_type: 2
    emergency_shutdown_enable: true
    address:
      street: ""
      city: ""
      zipcode: ""
      housenumber: ""
      country: Germany
      country_code: DE
      additionalInfo: ""
      state: ""
    rooms:
      - id: 3409538
        name: Küche
        type: 0
        room_type: 15
        role: owner
        appliances:
          - appliance_id: a1aa1111-11a1-111a-1111-a1a1111111aa
            installation_date: "2024-07-06T21:56:04.000+02:00"
            name: My GROHE Blue Home
            serial_number: 123456789456123456789456123456789
            type: 104
            version: 01.04.Z10.0300.0104
            tdt: "2024-12-10T20:50:03.000+01:00"
            timezone: 120
            config:
              thresholds: null
              measurement_period: null
              action_on_major_leakage: null
              action_on_minor_leakage: null
              action_on_micro_leakage: null
              monitor_frost_alert: null
              monitor_lower_flow_limit: null
              monitor_upper_flow_limit: null
              monitor_lower_pressure_limit: null
              monitor_upper_pressure_limit: null
              monitor_lower_temperature_limit: null
              monitor_upper_temperature_limit: null
              monitor_major_leakage: null
              monitor_minor_leakage: null
              monitor_micro_leakage: null
              monitor_system_error: null
              monitor_btw_0_1_and_0_8_leakage: null
              monitor_withdrawel_amount_limit_breach: null
              detection_interval: null
              impulse_ignore: null
              time_ignore: null
              pressure_tolerance_band: null
              pressure_drop: null
              detection_time: null
              action_on_btw_0_1_and_0_8_leakage: null
              action_on_withdrawel_amount_limit_breach: null
              withdrawel_amount_limit: null
              sprinkler_mode_active_monday: null
              sprinkler_mode_active_tuesday: null
              sprinkler_mode_active_wednesday: null
              sprinkler_mode_active_thursday: null
              sprinkler_mode_active_friday: null
              sprinkler_mode_active_saturday: null
              sprinkler_mode_active_sunday: null
              sprinkler_mode_start_time: null
              sprinkler_mode_stop_time: null
              measurement_transmission_intervall: null
              measurement_transmission_intervall_offset: null
              co2_type: 1
              hose_length: 90
              co2_consumption_medium: 48
              co2_consumption_carbonated: 65
              guest_mode_active: false
              auto_flush_active: false
              flush_confirmed: false
              f_parameter: 9
              l_parameter: 2
              flow_rate_still: 20
              flow_rate_medium: 25
              flow_rate_carbonated: 20
            role: owner
            registration_complete: true
            presharedkey: [secred]
            params:
              water_hardness: 0
              carbon_hardness: 6
              filter_type: 2
              variant: 4
              auto_flush_reminder_notif: true
              consumables_low_notif: true
              product_information_notif: true
            error:
              errors_1: false
              errors_2: false
              errors_3: false
              errors_4: false
              errors_5: false
              errors_6: false
              errors_7: false
              errors_8: false
              errors_9: false
              errors_10: false
              errors_11: false
              errors_12: false
              errors_13: false
              errors_14: false
              errors_15: false
              errors_16: false
              error1_counter: 0
              error2_counter: 1
              error3_counter: 67
              error4_counter: 0
              error5_counter: 0
              error6_counter: 0
              error7_counter: 0
              error8_counter: 2
              error9_counter: 0
              error10_counter: 0
              error11_counter: 0
              error12_counter: 121
              error13_counter: 0
              error14_counter: 0
              error15_counter: 0
              error16_counter: 0
            state:
              start_time: 1733860413
              APPLIANCE_SUCCESSFUL_CONFIGURED: false
              co2_empty: false
              co2_20l_reached: false
              filter_empty: false
              filter_20l_reached: false
              cleaning_mode_active: false
              cleaning_needed: false
              flush_confirmation_required: false
              System_error_bitfield: 0
            calculate_average_since: null
            pressure_notification: null
            snooze_status: null
            last_pressure_measurement: null
            installer: null
            command: null
            notifications:
              - appliance_id: a1aa1111-11a1-111a-1111-a1a1111111aa
                id: a1aa1111-11a1-111a-1111-a1a1111111aa
                category: 20
                is_read: false
                timestamp: "2024-12-06T20:04:38.000+01:00"
                type: 551
                threshold_quantity: "no"
                threshold_type: "no"
            status:
              - type: update_available
                value: 0
              - type: connection
                value: 1
            data_latest:
              measurement:
                timestamp: "2024-12-10T20:50:03.000+01:00"
                temperature: null
                humidity: null
                flow_rate: null
                pressure: null
                temperature_guard: null
                battery: null
                cleaning_count: 2
                date_of_cleaning: "2024-07-06T08:28:56.000+02:00"
                date_of_co2_replacement: "2024-11-24T15:27:30.000+01:00"
                date_of_filter_replacement: "2024-07-06T08:30:31.000+02:00"
                filter_change_count: 6
                max_idle_time: 21636
                open_close_cycles_carbonated: 15603
                open_close_cycles_still: 17068
                operating_time: 29951
                power_cut_count: 25
                pump_count: 26392
                pump_running_time: 2201
                remaining_co2: 18
                remaining_filter: 76
                time_since_last_withdrawal: 68
                time_since_restart: 536887329
                timeoffset: 120
                water_running_time_carbonated: 1831
                water_running_time_medium: 123
                water_running_time_still: 603
                remaining_filter_liters: 2290
                remaining_co2_liters: 10
              withdrawals: null
              average_monthly_consumption: null
              daily_cost: null
              average_daily_consumption: null
              daily_consumption: null`

If you need anymore information I'm glad to help.

**Comment by Flo-Schilli** [comment]
Hi @weaseel 👋

thanks for the info. I'm sorry to hear that you're having problems with the integration.
For me it seems like Grohe is changing something on their interface as they also removed some parts for the Sense Guard.

Nevertheless I wanted to do a refactoring but it will take a few evenings and then it would be great if you could also test it.

If you want you can just try the feature/refactoring branch which should have fixed already a few things and there are a lot of actions included to get data via the UI in HA. Maybe that one is already working for now until I complete the refactoring.

**Comment by weaseel** [comment]
Hi. Thanks for the fast reply.
I will try to Test your branch. it’s not possible to Download it via hacs, correct?
Additionally im currently trying to setup a dev Environment on my pc. but im currently stuck on that…
I already started a Python Implementation of the grohe sense. But then i found your component.

**Comment by Flo-Schilli** [comment]
Hey,

no unfortunately not. You have to check it out with git and then transfer it to your instance.
If you need assistance with setting up the dev environment for HA, the documentation from HA is good, but was for instance missing my PyCharm IDE.

**Comment by weaseel** [comment]
I had it Running once. But since i updated to the current versions it’s not working anymore. As soon as i have time I will try again

**Comment by weaseel** [comment]
feature/refactoring is also not working:
`  File "/workspaces/core/config/custom_components/grohe_sense/__init__.py", line 31, in async_setup_entry
    await api.login(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
  File "/workspaces/core/config/custom_components/grohe_sense/api/ondus_api.py", line 288, in login
    cookie, action = await self.__get_oidc_action()
    ^^^^^^^^^^^^^^
TypeError: cannot unpack non-iterable NoneType object`

**Comment by Flo-Schilli** [comment]
Hi @weaseel :wave: 

I just released a new Beta version which can be installed via HACS when choosing a different version. Would be great if you could give it a try as it should solve the problems :smile:

**Comment by weaseel** [comment]
I just installed it. Looks like the Old Version somehow started to work since I have Old values from a few days ago. There are still a lot of „unavailable“ values. I will post a list of it later.

**Comment by weaseel** [comment]
Here is the List. Thanks a lot for the Support. 
Cleaning_count
Date_of_cleaning
Date_of_co2_replacement
Date_of_filter_replacement
Filter_change_count
Max_idle_time
Notifications
Open_close_cycles_carbonated
Open_close_cycles_still
Operating_time
Power_cut_count
Pump_count
Pump_running_time
Remaining_co2_liters
Remaining_filter_liters
Water_running_time_carbonated
Water_running_time_medium
Water_running_time_still

I just realized you are also German. Of you want we can also Communicate outside of GitHub.

**Comment by Flo-Schilli** [comment]
Hi,

can you please run the following action and past the output here as a yaml code.
Just past your device name inside and push the perform action.

![image](https://github.com/user-attachments/assets/912dc7e6-3120-4b3d-bc0c-6ca22dacc841)

Also - if above mentioned values are still unavailable, can you please enable the debug logging for the integration in the configuration yaml like this:

```yaml
logger:
  default: info
  logs:
    custom_components.grohe_sense: debug
```

You should then see something like this in the logs:

```
2024-12-17 10:25:42.038 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Water consumption" has the following value on keypath "details.data_latest.daily_consumption": 56
2024-12-17 10:25:42.045 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Average daily consumption" has the following value on keypath "details.data_latest.average_daily_consumption": 345
2024-12-17 10:25:42.051 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Latest consumption" has the following value on keypath "details.data_latest.withdrawals.waterconsumption": 0.7
2024-12-17 10:25:42.058 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Latest max flow rate" has the following value on keypath "details.data_latest.withdrawals.maxflowrate": 5
2024-12-17 10:25:42.065 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Average monthly consumption" has the following value on keypath "details.data_latest.average_monthly_consumption": 10335
2024-12-17 10:25:42.072 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Actual flow rate" has the following value on keypath "details.data_latest.measurement.flowrate": 0
2024-12-17 10:25:42.078 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Actual pressure" has the following value on keypath "details.data_latest.measurement.pressure": 3.5
2024-12-17 10:25:42.084 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Latest notification" has the following value on keypath "details.notifications[0]": Potential leak detected
2024-12-17 10:25:42.090 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "PM: Pressure drop" has the following value on keypath "pressure.items[0].drop_of_pressure": -2.5
2024-12-17 10:25:42.097 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "PM: Is Leakage" has the following value on keypath "pressure.items[0].leakage": True
2024-12-17 10:25:42.103 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "PM: Leakage level" has the following value on keypath "pressure.items[0].level": 5
2024-12-17 10:25:42.109 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "PM: Runtime" has the following value on keypath "pressure.items[0].start_time": 2024-12-17 05:44:00+01:00
2024-12-17 10:25:42.115 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: SenseGuard () with sensor name: "Temperature" has the following value on keypath "details.data_latest.measurement.temperature_guard": 18
```

That information would also be interessting :smile:

**Comment by weaseel** [comment]
Looks like everything is there
`appliance_id: d8ea0672-28e5-418c-8521-e7f3430025bb
installation_date: "2024-07-06T21:56:04.000+02:00"
name: My GROHE Blue Home
serial_number: ABCDEFGSECRET
type: 104
version: 01.04.Z10.0300.0104
tdt: "2024-12-17T13:57:31.000+01:00"
timezone: 120
config:
  co2_type: 1
  co2_consumption_medium: 48
  co2_consumption_carbonated: 65
  guest_mode_active: false
  auto_flush_active: false
  flush_confirmed: false
  hose_length: 90
  f_parameter: 9
  l_parameter: 2
  flow_rate_still: 20
  flow_rate_medium: 25
  flow_rate_carbonated: 20
  cleaningPeriod1: false
  cleaningPeriod2: false
role: owner
registration_complete: true
presharedkey: ABCDEFGSECRET
params:
  water_hardness: 0
  carbon_hardness: 6
  filter_type: 2
  variant: 4
  auto_flush_reminder_notif: true
  consumables_low_notif: true
  product_information_notif: true
error:
  errors_1: false
  errors_2: false
  errors_3: false
  errors_4: false
  errors_5: false
  errors_6: false
  errors_7: false
  errors_8: false
  errors_9: false
  errors_10: false
  errors_11: false
  errors_12: false
  errors_13: false
  errors_14: false
  errors_15: false
  errors_16: false
  error1_counter: 0
  error2_counter: 1
  error3_counter: 67
  error4_counter: 0
  error5_counter: 0
  error6_counter: 0
  error7_counter: 0
  error8_counter: 2
  error9_counter: 0
  error10_counter: 0
  error11_counter: 0
  error12_counter: 122
  error13_counter: 0
  error14_counter: 0
  error15_counter: 0
  error16_counter: 0
state:
  start_time: 1734441278
  APPLIANCE_SUCCESSFUL_CONFIGURED: false
  co2_empty: false
  co2_20l_reached: false
  filter_empty: false
  filter_20l_reached: false
  cleaning_mode_active: false
  cleaning_needed: false
  flush_confirmation_required: false
  System_error_bitfield: 0
status:
  - type: connection
    value: 1
  - type: update_available
    value: 0
data_latest:
  measurement:
    cleaning_count: 2
    date_of_cleaning: "2024-07-06T08:28:56.000+02:00"
    date_of_co2_replacement: "2024-12-14T16:18:50.000+01:00"
    date_of_filter_replacement: "2024-07-06T08:30:31.000+02:00"
    filter_change_count: 6
    max_idle_time: 21636
    open_close_cycles_carbonated: 15684
    open_close_cycles_still: 17194
    operating_time: 30112
    power_cut_count: 25
    pump_count: 26552
    pump_running_time: 2213
    remaining_co2: 79
    remaining_filter: 75
    time_since_last_withdrawal: 3
    time_since_restart: 536887329
    timeoffset: 120
    timestamp: "2024-12-17T13:57:31.000+01:00"
    water_running_time_carbonated: 1841
    water_running_time_medium: 124
    water_running_time_still: 608
    remaining_filter_liters: 2258
    remaining_co2_liters: 44
notifications: []
`
Btw: If I remember correctly, I was able to even dispense some water with the rest API. I check if I can find the commands that I used and create another issue.

**Comment by Flo-Schilli** [comment]
But still all of the sensors are unavailable on your side even the above mentioned details are returned? 
If so, what are the logs saying 😄 
By the way, I added some general contact infos to my profile 😊 

That would be great if you could check that commands for dispensing the water. I guess one day I also need to buy one just for testing 🤣

**Comment by weaseel** [comment]
Ah and here are the logs:
```
2024-12-17 15:32:39.436 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Latest notification for device My GROHE Blue Home
2024-12-17 15:32:39.437 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Cycles carbonated for device My GROHE Blue Home
2024-12-17 15:32:39.438 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Cycles still for device My GROHE Blue Home
2024-12-17 15:32:39.438 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Date cleaning for device My GROHE Blue Home
2024-12-17 15:32:39.439 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Date CO2 replacement for device My GROHE Blue Home
2024-12-17 15:32:39.440 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Date Filter replacement for device My GROHE Blue Home
2024-12-17 15:32:39.441 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time idle (max) for device My GROHE Blue Home
2024-12-17 15:32:39.441 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time operating for device My GROHE Blue Home
2024-12-17 15:32:39.442 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time pump running for device My GROHE Blue Home
2024-12-17 15:32:39.443 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time since last withdrawal for device My GROHE Blue Home
2024-12-17 15:32:39.443 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Water running carbonated for device My GROHE Blue Home
2024-12-17 15:32:39.444 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Water running medium for device My GROHE Blue Home
2024-12-17 15:32:39.445 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Water running still for device My GROHE Blue Home
2024-12-17 15:32:39.445 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Remaining CO2 for device My GROHE Blue Home
2024-12-17 15:32:39.446 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Remaining Filter for device My GROHE Blue Home
2024-12-17 15:32:39.446 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Remaining CO2 total for device My GROHE Blue Home
2024-12-17 15:32:39.447 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Remaining Filter total for device My GROHE Blue Home
2024-12-17 15:32:39.447 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Count cleaning for device My GROHE Blue Home
2024-12-17 15:32:39.448 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Count filter change for device My GROHE Blue Home
2024-12-17 15:32:39.449 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Count Power cut for device My GROHE Blue Home
2024-12-17 15:32:39.449 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Count pump for device My GROHE Blue Home
2024-12-17 15:32:39.466 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time since restart for device My GROHE Blue Home
2024-12-17 15:32:39.466 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Time offset for device My GROHE Blue Home
2024-12-17 15:32:39.467 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Update available for device My GROHE Blue Home
2024-12-17 15:32:39.468 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity_helper] Adding sensor Online for device My GROHE Blue Home
2024-12-17 15:32:39.468 DEBUG (MainThread) [custom_components.grohe_sense.entities.coordinator.blue_home_coordinator] Updating device data for device GroheTypes.GROHE_BLUE_HOME with name My GROHE Blue Home (appliance = ABC-DEF-GHI)
2024-12-17 15:32:39.468 DEBUG (MainThread) [custom_components.grohe_sense.api.ondus_api] Get appliance details for appliance (type insensitive) ABC-DEF-GHI
2024-12-17 15:32:40.373 DEBUG (MainThread) [custom_components.grohe_sense.entities.coordinator.blue_home_coordinator] Finished fetching Grohe Sense data in 0.905 seconds (success: True)
2024-12-17 15:32:40.374 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Cycles carbonated" has the following value on keypath "details.data_latest.measurement.open_close_cycles_carbonated": 15685
2024-12-17 15:32:40.375 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Cycles still" has the following value on keypath "details.data_latest.measurement.open_close_cycles_still": 17197
2024-12-17 15:32:40.375 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Date cleaning" has the following value on keypath "details.data_latest.measurement.date_of_cleaning": 2024-07-06 08:28:56+02:00
2024-12-17 15:32:40.376 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Date CO2 replacement" has the following value on keypath "details.data_latest.measurement.date_of_co2_replacement": 2024-12-14 16:18:50+01:00
2024-12-17 15:32:40.377 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Date Filter replacement" has the following value on keypath "details.data_latest.measurement.date_of_filter_replacement": 2024-07-06 08:30:31+02:00
2024-12-17 15:32:40.377 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time idle (max)" has the following value on keypath "details.data_latest.measurement.max_idle_time": 21636
2024-12-17 15:32:40.378 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time operating" has the following value on keypath "details.data_latest.measurement.operating_time": 30113
2024-12-17 15:32:40.379 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time pump running" has the following value on keypath "details.data_latest.measurement.pump_running_time": 2213
2024-12-17 15:32:40.379 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time since last withdrawal" has the following value on keypath "details.data_latest.measurement.time_since_last_withdrawal": 16
2024-12-17 15:32:40.380 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Water running carbonated" has the following value on keypath "details.data_latest.measurement.water_running_time_carbonated": 1841
2024-12-17 15:32:40.381 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Water running medium" has the following value on keypath "details.data_latest.measurement.water_running_time_medium": 124
2024-12-17 15:32:40.381 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Water running still" has the following value on keypath "details.data_latest.measurement.water_running_time_still": 608
2024-12-17 15:32:40.382 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Remaining CO2" has the following value on keypath "details.data_latest.measurement.remaining_co2": 77
2024-12-17 15:32:40.382 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Remaining Filter" has the following value on keypath "details.data_latest.measurement.remaining_filter": 75
2024-12-17 15:32:40.384 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Remaining CO2 total" has the following value on keypath "details.data_latest.measurement.remaining_co2_liters": 43
2024-12-17 15:32:40.384 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Remaining Filter total" has the following value on keypath "details.data_latest.measurement.remaining_filter_liters": 2257
2024-12-17 15:32:40.385 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Count cleaning" has the following value on keypath "details.data_latest.measurement.cleaning_count": 2
2024-12-17 15:32:40.386 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Count filter change" has the following value on keypath "details.data_latest.measurement.filter_change_count": 6
2024-12-17 15:32:40.386 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Count Power cut" has the following value on keypath "details.data_latest.measurement.power_cut_count": 25
2024-12-17 15:32:40.387 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Count pump" has the following value on keypath "details.data_latest.measurement.pump_count": 26555
2024-12-17 15:32:40.388 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time since restart" has the following value on keypath "details.data_latest.measurement.time_since_restart": 536887329
2024-12-17 15:32:40.388 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Time offset" has the following value on keypath "details.data_latest.measurement.timeoffset": 120
2024-12-17 15:32:40.389 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Update available" has the following value on keypath "status.update_available": 0
2024-12-17 15:32:40.389 DEBUG (MainThread) [custom_components.grohe_sense.entities.entity.sensor] Device: My GROHE Blue Home (ABC-DEF-GHI) with sensor name: "Online" has the following value on keypath "status.connection": 1
```

**Comment by Flo-Schilli** [comment]
So all values are now okay and not unknown/unavailable anymore, right? :smile:

**Comment by Flo-Schilli** [comment]
I added with #48 two more actions for the integration, so that you should be able to set the commands for yourself. They should also be available in automations.

First you need to get the commands available with the following action:
![image](https://github.com/user-attachments/assets/9ee28550-a00c-4da0-9bb9-fe39f42ba632)


and then you can set it like this and try if it is working:
![image](https://github.com/user-attachments/assets/ebb05b52-c57c-4624-9cc7-2bdebbb75d36)

**Comment by weaseel** [comment]
> So all values are now okay and not unknown/unavailable anymore, right? 😄

No still unavailable

**Comment by Flo-Schilli** [comment]
Can you send me a screenshot please. In regards to the logs everything is fine and set

**Comment by weaseel** [comment]
BTW i think the ones unavailabe are somehow old entities. They all exist twice.

**Comment by weaseel** [comment]
Btw it seems that the data is Not always Updated. In my Implemention i always send the „get_current_measurement“ command.
Line 85 (please ignore the ugly code :P):
https://github.com/weaseel/groheondus/blob/main/src/GroheOndus.py

You can see the statistics here. Before 9am i was Running my own Script with the „get_current_measurement“ command. 
the values shows how long no water was drawn. 
![image](https://github.com/user-attachments/assets/82f69631-db90-4fa9-9264-110e272d4450)

**Comment by weaseel** [comment]
Tried this but it didnt work. 
![image](https://github.com/user-attachments/assets/d934dd63-dc3e-42ca-af05-32edc0a11e6f)

logs seem to be Fine. 
```
2024-12-17 19:40:41.482 DEBUG (MainThread) [custom_components.grohe_sense] Set commands for params: {'commands': {'command': {'co2_status_reset': False, 'tap_type': 2, 'cleaning_mode': False, 'filter_status_reset': False, 'get_current_measurement': False, 'tap_amount': 50, 'factory_reset': False, 'revoke_flush_confirmation': False, 'exec_auto_flush': False}}, 'device_name': 'My GROHE Blue Home'}
2024-12-17 19:40:54.129 DEBUG (MainThread) [custom_components.grohe_sense] Set commands for params: {'commands': {'command': {'co2_status_reset': False, 'tap_type': 1, 'cleaning_mode': False, 'filter_status_reset': False, 'get_current_measurement': False, 'tap_amount': 100, 'factory_reset': False, 'revoke_flush_confirmation': False, 'exec_auto_flush': False}}, 'device_name': 'My GROHE Blue Home'}
```

btw i contacted you via mastodon

**Comment by Flo-Schilli** [comment]
> Tried this but it didnt work. 
> ![image](https://github.com/user-attachments/assets/d934dd63-dc3e-42ca-af05-32edc0a11e6f)
> 
> logs seem to be Fine. 
> ```
> 2024-12-17 19:40:41.482 DEBUG (MainThread) [custom_components.grohe_sense] Set commands for params: {'commands': {'command': {'co2_status_reset': False, 'tap_type': 2, 'cleaning_mode': False, 'filter_status_reset': False, 'get_current_measurement': False, 'tap_amount': 50, 'factory_reset': False, 'revoke_flush_confirmation': False, 'exec_auto_flush': False}}, 'device_name': 'My GROHE Blue Home'}
> 2024-12-17 19:40:54.129 DEBUG (MainThread) [custom_components.grohe_sense] Set commands for params: {'commands': {'command': {'co2_status_reset': False, 'tap_type': 1, 'cleaning_mode': False, 'filter_status_reset': False, 'get_current_measurement': False, 'tap_amount': 100, 'factory_reset': False, 'revoke_flush_confirmation': False, 'exec_auto_flush': False}}, 'device_name': 'My GROHE Blue Home'}
> ```
> 
> btw i contacted you via mastodon

You just need to send the real needed commands like:

```yaml
co2_status_reset: false
```

Then it should work 😊

**Comment by weaseel** [comment]
I tried to get some water from it. Somehow it still doesnt work. 
![image](https://github.com/user-attachments/assets/6a1498c8-1ea8-4b13-a7d4-5c14cd0b706a)


Btw i found this repo https://github.com/koproductions-code/ha-groheblue
Maybe you two can cooperate

**Comment by Flo-Schilli** [comment]
> Btw it seems that the data is Not always Updated. In my Implemention i always send the „get_current_measurement“ command.
> Line 85 (please ignore the ugly code :P):
> https://github.com/weaseel/groheondus/blob/main/src/GroheOndus.py
> 
> You can see the statistics here. Before 9am i was Running my own Script with the „get_current_measurement“ command. 
> the values shows how long no water was drawn. 
> ![image](https://github.com/user-attachments/assets/82f69631-db90-4fa9-9264-110e272d4450)
> 

What's the sensor you're referring to? And I'm not sure about how the update for the blue home is done on the Grohe Side as I'm only using the details.data_latest.measurement values, so without triggering a new measurement

**Comment by Flo-Schilli** [comment]
> BTW i think the ones unavailabe are somehow old entities. They all exist twice. 
> 
> ![image](https://github.com/user-attachments/assets/c69ac155-a40e-41cb-b530-07a193ff7c36)
> 

Yeah the unavailable ones are definitely ones from before. As the names changed with the refactoring the other ones are still there but not provided anymore for everything starting with 0.20.XX

**Comment by Flo-Schilli** [comment]
> Btw it seems that the data is Not always Updated. In my Implemention i always send the „get_current_measurement“ command.
> Line 85 (please ignore the ugly code :P):
> https://github.com/weaseel/groheondus/blob/main/src/GroheOndus.py
> 
> You can see the statistics here. Before 9am i was Running my own Script with the „get_current_measurement“ command. 
> the values shows how long no water was drawn. 
> ![image](https://github.com/user-attachments/assets/82f69631-db90-4fa9-9264-110e272d4450)
> 

What's the sensor you're referring to? And I'm not sure about how the update for the blue home is done on the Grohe Side as I'm only using the details.data_latest.measurement values, so without triggering a new measurement.
Also it would have been better (from Grohe) to get that one as a timestamp than a total time from the backend.

Can you create for that topic a new issue just to not mix up everything in one thread 😊

**Comment by Flo-Schilli** [comment]
Released a new version (v0.21.0) which should address all the issues. If not please create a new ticket
